yt/frontends/boxlib/data_structures.py

```diff
@@ -713,7 +711,14 @@ def _is_valid_subtype(cls, *args, **kwargs):
         if not BoxlibDataset._is_valid(output_dir):
             return False
 
-        cparam_filepath = cls._lookup_cparam_filepath(*args, **kwargs)
+        try:
```

Comment:
OK, I have to admit, this seems extra magical, even for loading files.  Can you at least put some comments here justifying this incantation?

Reply:
yeah I'm not a fan either, now that I'm not struggling with it anymore.
I'll try to clarify this, thanks for pointing this out

Reply:
I came out with a better solution. I hope you like it better now !